pkg/awsManager/ec2Manager.go

```diff
@@ -0,0 +1,122 @@
+
+	var EC2InstancesToBeDeleted []*string
+	token := ""
+	for {
+		ec2Descriptions, err := assumedRoleClient.DescribeInstances(&ec2.DescribeInstancesInput{NextToken: aws.String(token)})
+		if err != nil {
+			fmt.Println("ERROR:", err)
+			os.Exit(1)
+		}
+
+		// nested for loop to read the tags , as it is a part of structure inside a structure output. Refer : https://pkg.go.dev/github.com/aws/aws-sdk-go/service/ec2?tab=doc#DescribeInstancesOutput
+		for _, iterator := range ec2Descriptions.Reservations {
+			for _, iterator2 := range iterator.Instances {
+				for _, iterator3 := range iterator2.Tags {
+
+					//checking if the tags start with 'kubernetes.io' and status code !=48
+					if strings.HasPrefix(*iterator3.Key, "kubernetes.io") && (*iterator2.State.Code != 48) {
+						// some debugging statements , will be removed after thorough testing.
+						//	fmt.Println(*iterator2.InstanceId)
+						//	fmt.Println(*iterator3.Key)
+						//	fmt.Println(*iterator3.Value)
+						//	fmt.Println(iterator2.State)
+						//	fmt.Println("_______________________________________________________________________________")
+						EC2InstancesToBeDeleted = append(EC2InstancesToBeDeleted, iterator2.InstanceId)
+						break
+					}
+
+					if (*iterator3.Key == "clusterAccountName" || *iterator3.Key == "clusterClaimLink" || *iterator3.Key == "clusterNamespace" || *iterator3.Key == "clusterClaimLinkNamespace") && (*iterator2.State.Code != 48) {
+						// some debugging statements , will be removed after thorough testing.
+						//	fmt.Println(*iterator2.InstanceId)
+						//	fmt.Println(*iterator3.Key)
+						//	fmt.Println(*iterator3.Value)
+						//	fmt.Println(iterator2.State)
+						//	fmt.Println("_______________________________________________________________________________")
+						EC2InstancesToBeDeleted = append(EC2InstancesToBeDeleted, iterator2.InstanceId)
+
+						break
+					}
+					// no need to delete the instances if the above two 'if' conditions are not true
+
+				}
+			}
+		}
+
+		// for pagination
+		if ec2Descriptions.NextToken != nil {
+			token = *ec2Descriptions.NextToken
+		} else {
+			break
+		}
+	}
+
+	return EC2InstancesToBeDeleted
+}
+
+// this takes all the instances from insancesToBeDeleted  ( one by one ) and deletes them
+func DeleteEc2Instance(assumedRoleClient clientpkg.Client, EC2InstancesToBeDeleted []*string) {
+
+	if EC2InstancesToBeDeleted == nil {
+		return
+	}
+
+	result, err := assumedRoleClient.TerminateInstances(&ec2.TerminateInstancesInput{InstanceIds: EC2InstancesToBeDeleted})
+	if err != nil {
+		if err, ok := err.(awserr.Error); ok {
+			switch err.Code() {
+			default:
+				fmt.Println(err.Error())
+			}
+		} else {
+			fmt.Println(err.Error())
+		}
+	}
+
+	// for debugging , will be removed afterwards
+	fmt.Println(result)
+
+}
+
+func CleanEc2Instacnes(assumedRoleClient clientpkg.Client) {
+
+	EC2InstancesToBeDeleted := ListEc2InstancesForDeletion(assumedRoleClient)
```

Comment:
`ec2InstancesToBeDeleted` not a big deal, but in Go any var that starts with a capital letter is exported. We should avoid that when possible.

Reply:
resolved.